kajas/src/app/registration/registration.component.html

```diff
@@ -0,0 +1,123 @@
+<div class="main-container">
+  <div class="container">
+    <div class="logo-mobile">
+      <img src="./assets/kajas_icon.png" alt="Kajas Logo">
+    </div>
+    <h1>Sign Up</h1>
+    <form [formGroup]="registrationForm" (ngSubmit)="onSubmit()">
+      <div class="form-group-wrapper">
+        <div class="form-group">
+          <input type="text" id="username" formControlName="username" 
+            placeholder="Username *" required>
+              @if(registrationForm.get('username').invalid && 
+                registrationForm.get('username').touched) {
+                <div class="error-message">
+                  {{ getErrorMessage('username') }}
+                </div>
+              }
+        </div>
+      </div>
+      <div class="form-group-wrapper">
+        <div class="form-group">
+          <input type="email" id="email" formControlName="email" 
+            placeholder="Email Address *" required>
+              @if(registrationForm.get('email').invalid && 
+                registrationForm.get('email').touched) {
+                  <div class="error-message">
+                    {{ getErrorMessage('email') }}
+                  </div>
+              }
+        </div>
+      </div>
+      <div class="form-group-wrapper password-container">
+        <div class="form-group">
+          <input [type]="hidePassword ? 'password' : 'text'" id="password" 
+            formControlName="password" placeholder="Password *" required>
+          <img src="./assets/icons/{{hidePassword ? 'unsee' : 'see'}}.png" 
+            alt="Toggle Password Visibility" class="toggle-password"
+            (click)="togglePasswordVisibility('password')">
+              @if(registrationForm.get('password').invalid && 
```

Comment:
### For template control flow, observe proper indentation
- Try to find a [solution](https://github.com/angular/angular/issues/52720) online or uninstall prettier and format your codes manually.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_15